cmake/toolchain-gcc.cmake

```diff
@@ -1,15 +1,15 @@
 # Configures CMake for using GCC, this script is re-used by several
 # GCC-based toolchains
 
-set(CMAKE_C_COMPILER   ${CROSS_COMPILE}gcc     CACHE INTERNAL " " FORCE)
-set(CMAKE_OBJCOPY      ${CROSS_COMPILE}objcopy CACHE INTERNAL " " FORCE)
-set(CMAKE_OBJDUMP      ${CROSS_COMPILE}objdump CACHE INTERNAL " " FORCE)
-#set(CMAKE_LINKER      ${CROSS_COMPILE}ld      CACHE INTERNAL " " FORCE) # Not in use yet
-set(CMAKE_AR           ${CROSS_COMPILE}ar      CACHE INTERNAL " " FORCE)
-set(CMAKE_RANLILB      ${CROSS_COMPILE}ranlib  CACHE INTERNAL " " FORCE)
-set(CMAKE_READELF      ${CROSS_COMPILE}readelf CACHE INTERNAL " " FORCE)
-set(CMAKE_GDB          ${CROSS_COMPILE}gdb     CACHE INTERNAL " " FORCE)
-set(CMAKE_NM           ${CROSS_COMPILE}nm      CACHE INTERNAL " " FORCE)
+find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}gcc     PATH ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
```

Comment:
I know that `TOOLCHAIN_HOME` lacks the `/bin` suffix but it somehow works on both Linux and Windows

Reply:
@carlescufi : Did you try omitting TOOLCHAIN_HOME?

Looks like CROSS_COMPILE includes TOOLCHAIN_HOME.

Reply:
Yeah, I did and it worked. But I didn't want to do that because I wanted to explicitly use `NO_DEFAULT_PATH` to ensure we only look at a particular path, defined by `TOOLCHAIN_HOME`

Reply:
Okay, so you could only omit CROSS_COMPILE when you also omitted NO_DEFAULT_PATH?

As it is now I don't think find_program is using the path defined by TOOLCHAIN_HOME because CROSS_COMPILE cannot be found in TOOLCHAIN_HOME.

If this works it doesn't matter I guess, we can clean this up at some other time, e.g. when we resolve #5517.

Reply:
>Okay, so you could only omit CROSS_COMPILE when you also omitted NO_DEFAULT_PATH?

I think so, yes

>As it is now I don't think find_program is using the path defined by TOOLCHAIN_HOME because CROSS_COMPILE cannot be found in TOOLCHAIN_HOME

Correct, but it's no big deal. It works on both Windows and Linux, so I'm not worried.